src/Core/Data/TripleBuffer.hpp

```
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <atomic>
+#include <type_traits>
+
+
+/* Triple-buffer implementation for Producer-Consumer-pattern systems. */
+template<typename T>
+class TripleBuffer {
+public:
+    TripleBuffer() : m_writeIndex(0), m_readIndex(1), m_dirtyIndex(2) {
```

Clang (14.0.6)

[warning] constructor does not initialize these fields: m_dirtyIndex

(cppcoreguidelines-pro-type-member-init)